internal/graph/check_test.go

```diff
@@ -996,6 +996,230 @@ condition condition1(param1: string) {
+				User:     "group:1#member",
+				Relation: "member",
+				Object:   "group:eng",
+			},
+			{
+				User:     "group:2#member",
+				Relation: "member",
+				Object:   "group:eng",
+			},
+			{
+				User:     "group:3#member",
+				Relation: "member",
+				Object:   "group:eng",
+			},
+			{
+				User:     "group:eng#member",
+				Relation: "viewer",
+				Object:   "document:1",
+			},
+		})
+		require.NoError(t, err)
+
+		checker := NewLocalChecker()
+
+		typesys, err := typesystem.NewAndValidate(
+			context.Background(),
+			model,
+		)
+		require.NoError(t, err)
+
+		ctx = typesystem.ContextWithTypesystem(ctx, typesys)
+
+		resp, err := checker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:              storeID,
+			AuthorizationModelID: model.GetId(),
+			TupleKey:             tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+			ResolutionMetadata:   &ResolutionMetadata{Depth: 5},
+		})
+		require.NoError(t, err)
+		require.True(t, resp.Allowed)
+
+		require.GreaterOrEqual(t, resp.GetResolutionMetadata().DispatchCount, uint32(2))
+		require.LessOrEqual(t, resp.GetResolutionMetadata().DispatchCount, uint32(4))
+
+		resp, err = checker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:              storeID,
+			AuthorizationModelID: model.GetId(),
+			TupleKey:             tuple.NewTupleKey("document:1", "viewer", "user:other"),
+			ResolutionMetadata:   &ResolutionMetadata{Depth: 5},
+		})
+		require.NoError(t, err)
+		require.False(t, resp.Allowed)
+
+		require.Equal(t, uint32(4), resp.GetResolutionMetadata().DispatchCount)
+	})
+
+	t.Run("dispatch_count_computed_userset_lookups", func(t *testing.T) {
+		storeID := ulid.Make().String()
+
+		model := parser.MustTransformDSLToProto(`model
+		schema 1.1
+
+		type user
+  	
+		type document
+			relations
+		   		define owner: [user]
+		   		define editor: [user] or owner`)
+
+		err := ds.Write(ctx, storeID, nil, []*openfgav1.TupleKey{
+			{
+				User:     "user:jon",
+				Relation: "owner",
+				Object:   "document:1",
+			},
+			{
+				User:     "user:will",
+				Relation: "editor",
+				Object:   "document:2",
+			},
+		})
+		require.NoError(t, err)
+
+		checker := NewLocalChecker()
+
+		typesys, err := typesystem.NewAndValidate(
+			context.Background(),
+			model,
+		)
+		require.NoError(t, err)
+
+		ctx = typesystem.ContextWithTypesystem(ctx, typesys)
+
+		resp, err := checker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:              storeID,
+			AuthorizationModelID: model.GetId(),
+			TupleKey:             tuple.NewTupleKey("document:1", "owner", "user:jon"),
+			ResolutionMetadata:   &ResolutionMetadata{Depth: 5},
+		})
+		require.NoError(t, err)
+		require.True(t, resp.Allowed)
+
+		require.Zero(t, resp.GetResolutionMetadata().DispatchCount)
+
+		resp, err = checker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:              storeID,
+			AuthorizationModelID: model.GetId(),
+			TupleKey:             tuple.NewTupleKey("document:2", "editor", "user:will"),
```

Comment:
And `ResolveCheck(document:1#editor@user:jon)` should always require exactly 1 dispatch (because `user:jon` is a `document:1#owner`). It may be good to just add that one as well. The `user:will` example below covers the overlap with the race on `editor or owner`. But for `user:jon` it should always be deterministic with 1 dispatch.